core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -1010,6 +1011,39 @@ private void basePushFilterPastAggWithGroupingSets(boolean unchanged)
     sql(sql).with(program).check();
   }
 
+  @Test void testSortRemoveRowCountBased() {
+    final HepProgram program = new HepProgramBuilder()
+        .addRuleInstance(SortRemoveRule.INSTANCE)
+        .build();
+    final String sql = "select count(*) as c\n"
+        + "from sales.emp\n"
+        + "where deptno = 10\n"
+        + "limit 10";
+    sql(sql).with(program).check();
+  }
+
+  @Test void testSortRemoveRowCountBasedZeroLimit() {
+    final HepProgram program = new HepProgramBuilder()
+        .addRuleInstance(SortRemoveRule.INSTANCE)
```

Comment:
this test actually only tests the `RelBuilder`; and it could pass without the rule added...
it's harder to test stuff when optimizations are being done in `RelBuilder`....

Reply:
@kgyrtkirk Do you mean this particular test or all tests added in `RelOptRulesTest` in this pull request? I quickly checked other tests by removing the rule but they fail so looks like `RelBuilder` is not being used.

Reply:
only 2 of them...I think we might not be able to check core features of some rules because the RelBuiilder may fix it beforehand.

at some point we should either have to introduce a switch to disable `RelBuilder` being smart...or make the current one a derivative of a basic one.

but anyway...this is outside of the scope of this change - I'll file a ticket for it